examples/drawing/plot_rainbow_coloring.py

```diff
@@ -0,0 +1,58 @@
+"""
+================
+Rainbow Coloring
+================
+
+Generate a complete graph with `n` nodes in a circular layout with the
+edges colored by node distance. The node distance is given by the minimum
+number of nodes traversed along an arc between any two nodes on the circle.
+"""
+import matplotlib.pyplot as plt
+import networkx as nx
+
+# A rainbow color mapping using matplotlib's tableau colors
+node_dist_to_color = {
+    1: "tab:red",
+    2: "tab:orange",
+    3: "tab:olive",
+    4: "tab:green",
+    5: "tab:blue",
+    6: "tab:purple",
```

Comment:
Maybe use a color map instead so the user can change the number of nodes and still have it work?

Reply:
I had considered this as well, but ultimately decided that the explicit 1->1 distance-to-color mapping was more in keeping with the example, even though it makes it less generalized. 

I'm happy to change it up though if there is a strong preference for the cmap-sampling approach.